src/idl/src/keylist.c

```diff
@@ -255,26 +327,26 @@ static idl_retcode_t get_keylist_key_paths(idl_pstate_t *pstate, void *list, str
 idl_retcode_t idl_validate_keylists(idl_pstate_t *pstate)
 {
   idl_retcode_t ret = IDL_RETCODE_OK;
-  struct key_container *key_containers = NULL;
-  size_t nkc = 0;
+  struct key_containers *key_containers;
+  if ((key_containers = key_containers_new ()) == NULL)
+    return IDL_RETCODE_NO_MEMORY;
 
   assert(pstate);
-  if ((ret = get_keylist_key_paths(pstate, pstate->root, &key_containers, &nkc) != IDL_RETCODE_OK))
+  if ((ret = get_keylist_key_paths (pstate, pstate->root, key_containers) != IDL_RETCODE_OK))
     goto err_create_key;
+  sort_parent_paths (key_containers);
```

Comment:
`has_conflicting_keys` requires the parent paths to be sorted, so wouldn't it be better to move this call to that function?

Reply:
The rationale for doing it this way was that this allows `has_conflicting_keys` to take a `const` input. The desire for that has to do with the name: `has_...` to me suggests a predicate, not a procedure that (potentially) modifies its input. So either we leave it this way, or we have to come up with another for `has_conflicting_keys`. How would you feel about `check_for_conflicting_keys`?

Reply:
Good point, I'm fine with the `check_` variant, or just a comment on the current `has_` function that it requires sorted input.

Reply:
Done!

@dpotman If you're happy with the updates I did based on your comments, then, after nearly half a year, I'll finally merge this PR 🙂

Reply:
Yes, I am, this PR can be merged!